scripts/e2e.mjs

```diff
@@ -0,0 +1,219 @@
+
+  // Set up platform-specific app configuration
+  if (platform == 'ios') {
+    execSync('ruby --version', { stdio: 'inherit', cwd: `${appDir}`, env: env });
+
+    // Fixes Hermes pod install https://github.com/CocoaPods/CocoaPods/issues/12226#issuecomment-1930604302
+    execSync(`gem install cocoapods -v 1.15.2`, { stdio: 'inherit', cwd: appDir, env: env });
+    execSync(`../../../../rn.patch.podfile.js --pod-file Podfile --engine ${RNEngine}`, { stdio: 'inherit', cwd: `${appDir}/ios`, env: env });
+
+    // const podInstallCommand = RNVersion === '0.65.3' ? 'pod install' : 'bundle exec pod install';
+    execSync('pod install --repo-update', { stdio: 'inherit', cwd: `${appDir}/ios`, env: env });
+    execSync('cat Podfile.lock | grep RNSentry', { stdio: 'inherit', cwd: `${appDir}/ios`, env: env });
+
+    execSync(`../../../rn.patch.xcode.js --project ios/${appName}.xcodeproj/project.pbxproj --rn-version ${RNVersion}`, { stdio: 'inherit', cwd: appDir, env: env });
+  } else if (platform == 'android') {
+    execSync(`../../../rn.patch.gradle.properties.js --gradle-properties android/gradle.properties --engine ${RNEngine}`, { stdio: 'inherit', cwd: appDir, env: env });
+    execSync(`../../../rn.patch.app.build.gradle.js --app-build-gradle android/app/build.gradle`, { stdio: 'inherit', cwd: appDir, env: env });
+
+    if (env.RCT_NEW_ARCH_ENABLED) {
+      execSync(`perl -i -pe's/newArchEnabled=false/newArchEnabled=true/g' android/gradle.properties`, { stdio: 'inherit', cwd: appDir, env: env });
+      console.log('New Architecture enabled');
+    }
+  }
+}
+
+if (actions.includes('build')) {
+  console.log(`Building ${platform}: ${buildType}`);
+  var appProduct;
+
+  if (platform == 'ios') {
+    const runtime = env.IOS_RUNTIME ? env.IOS_RUNTIME : 'latest';
+    const device = env.IOS_DEVICE ? env.IOS_DEVICE : 'iPhone 15';
+
+    // Build iOS test app
+    execSync(`set -o pipefail && xcodebuild \
+    -workspace ${appName}.xcworkspace \
+    -configuration ${buildType} \
+    -scheme ${appName} \
+    -destination 'platform=iOS Simulator,OS=${runtime},name=${device}' \
+    ONLY_ACTIVE_ARCH=yes \
+    -derivedDataPath DerivedData \
+    build | tee xcodebuild.log | xcbeautify`,
+      { stdio: 'inherit', cwd: `${appDir}/ios`, env: env });
+
+    appProduct = `${appDir}/ios/DerivedData/Build/Products/${buildType}-iphonesimulator/${appName}.app`;
+
+    // Build iOS WebDriverAgent
+    execSync(`set -o pipefail && xcodebuild \
+      -project node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj \
+      -scheme WebDriverAgentRunner \
+      -destination 'platform=iOS Simulator,OS=${runtime},name=${device}' \
+      GCC_TREAT_WARNINGS_AS_ERRORS=0 \
+      COMPILER_INDEX_STORE_ENABLE=NO \
+      ONLY_ACTIVE_ARCH=yes \
+      -derivedDataPath DerivedData \
+      build | tee xcodebuild-agent.log | xcbeautify`,
+      { stdio: 'inherit', cwd: e2eDir, env: env });
```

Comment:
I'm not sure if we can build and install the WebDriverAgent during the build phase. 

In CI the build is separate job from test, due to increased test time and flakiness when run in one job.

Because of the two jobs there is no guarrancy the build and test will run on the same runner (the same env, thus the webdriver might be missing)

Reply:
right

Reply:
moved to test with check on the file existing or not so it won't build multiple times when running locally.

Reply:
I agree it make sense during the build phase, but we would have to cache the DerivedData, which might be a good idea, since in between the tests the webdriver won't change.

Reply:
that's a good idea